lib/postgrex/connection.ex

```diff
@@ -85,16 +85,46 @@ defmodule Postgrex.Connection do
     message = {:query, statement, params}
     timeout = opts[:timeout] || @timeout
     case Connection.call(pid, message, timeout) do
-      %Postgrex.Result{} = res ->
-        {:ok, res}
-      %Postgrex.Error{} = err ->
-        {:error, err}
       {:exit, reason} ->
         exit({reason, {__MODULE__, :query, [pid, statement, params, opts]}})
+      result ->
+        result
     end
   end
 
   @doc """
+  Works like `query/3` and `query/4` but is asynchronous. This returns a `%Task{}`
+  and is useful for starting multiple async queries and then waiting on the result
+  later on.
+
+  When server name passed has no process associated, it will raise an error.
+
+  This function expects a pid as first argument for Elixir 1.0. In Elixir 1.1, a pid
+  or a name could be passed as first argument.
+
+  ## Examples
+
+      task = %Task{} = Postgrex.Connection.async_query(pid, "SELECT title FROM posts", [])
+      Task.await(task)
+  """
+  def async_query(pid, statement, params) do
```

Comment:
Can we get a spec on this and docs on the possible results of the task? Its important to document the `{:exit, reason}` result as it can't happen in query/3.